doc/md/guides/mysql/cd-101.md

```diff
@@ -0,0 +1,117 @@
+
+
+### Irreversible changes
+Dropping a foreign key constraint is not always a reversible operation. If data which would have been blocked by the constraint was inserted to the table, then the constraint cannot be re-applied without deleting the data.
+
+In the example above, let's insert some invalid data. assuming the `users` table is empty, we'll run:
+
+```sql
+INSERT INTO pets (name, owner_id) VALUES ('Fortuna', 1);
+```
+
+The, trying to re-create the constraint with:
+```sql
+ALTER TABLE pets ADD CONSTRAINT owner_id FOREIGN KEY (owner_id) REFERENCES users (id);
+```
+
+Will result in the following error:
+
+> [1452] Cannot add or update a child row: a foreign key constraint fails (\`pets\`.\`#sql-1_1140\`, CONSTRAINT \`owner_id\` FOREIGN KEY (\`owner_id\`) REFERENCES \`users\` (\`id\`))
+
+
+The only way the constraint could be applied again is by deleting all the invalid rows:
+```sql
+DELETE FROM pets WHERE name='Fortuna';
+```
+After running the last command, trying to add the constraint again will succeed.
+
+### Cascading
+Cascading policies on foreign keys may be used for various reasons. For example, a deletion cascading can help saving storage space by automatically cleaning irrelevant ("orphan") data. Dropping a foreign key constraint also cancels all cascading policies, and thus can cause unwanted behaviours as side effect of the constraint drop itself.
+
+## Prevention
+
+Preventing accidental constraint drop is easy with Atlas's [`migrate lint`](/versioned/lint)
+command. With the `atlas migrate lint` command, users can analyze the migration directory to 
+detect potentially dangerous changes to the database schema. This command may be 
+incorporated in continuous integration pipelines to enable teams to enforce 
+desired policies with regard to schema changes.
+
+When using migrate lint to analyze migrations, users must supply multiple parameters:
+
+* `--dev-url` - a URL to a [dev database](/concepts/dev-database) that will be used to simulate the changes and verify their correctness.
+* `--dir` - the URL of the migration directory, by default it is file://migrations, e.g a directory named migrations in the current working directory.
+
+### Changeset detection
+
+When we run the lint command, we need to instruct Atlas on how to decide what set of migration files to analyze. 
+Currently, two modes are supported:
+
+* `--git-base <branchName>`: selects the diff between the provided branch and the current one as the changeset.
+* `--latest <n>`: selects the latest n migration files as the changeset.
+
+### Examples
+
+Analyze all changes relative to the `master` Git branch:
+
+```bash
+atlas migrate lint \
+  --dir "file://my/project/migrations" \
+  --dev-url "mysql://root:pass@localhost:3306/dev" \
+  --git-base "master"
+```
+
+Analyze the latest two migration files:
+
+```bash
+atlas migrate lint \
+  --dir "file://my/project/migrations" \
+  --dev-url "mysql://root:pass@localhost:3306/dev" \
+  --latest 2
+```
```

Comment:
Remove from the guide please

Reply:
https://atlasgo.io/guides/mysql/checks/DS103#changeset-detection